src/components/MainContext/MainContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Errors } from '../../types/Errors';
+import { ContextProps } from '../../types/ContextProps';
+
+export const MainContext = React.createContext<ContextProps>({
+  users: [],
+  posts: [],
+  setPosts: () => { },
+  choosedPost: null,
+  setChoosedPost: () => { },
+  comments: [],
+  setComments: () => { },
+  error: Errors.NONE,
+  setError: () => { },
+  isLoading: false,
+  setIsLoading: () => { },
+  isActiveComForm: false,
+  setIsActiveComForm: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const MainProvider: React.FC<Props> = ({ children }) => {
```

Comment:
for the future it's much better to separate logic for different providers. Each provider should work with its own instance (one for `posts`, another for `user` etc)

Mentor Taras mentioned this as well 
https://github.com/mate-academy/react_dynamic-list-of-posts/pull/1026#discussion_r1469895823